ansible-scylla-node/tasks/ssl.yml

```diff
@@ -1,20 +1,33 @@
 ---
 - name: generate a self-signed CA and use it for node certs
   block:
+    - name: Check CA folder exists
+      stat:
+        path: ./ssl/ca
+      register: ca_folder_details
+
     - name: Create dir for the CA
       file:
         path: ./ssl/ca
         state: directory
+      when: ca_folder_details.stat.isdir == False
+
+    - name: Check CA pem exists
+      stat:
+        path: "./ssl/ca/{{ scylla_cluster_name }}-ca.crt"
+      register: ca_crt_details
 
     - name: Generate an OpenSSL private key for the CA.
```

Comment:
Why generate a private key if we tested the presence of a certificate. The key may exist already.

Reply:
That was my thought exactly, when I have ca.crt only it replaced it with self signed